controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -0,0 +1,98 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Reconciler is the reconciler for the SpaceProvisionerConfig CRs.
+type Reconciler struct {
+	Client runtimeclient.Client
+}
+
+var _ reconcile.Reconciler = (*Reconciler)(nil)
+
+func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.SpaceProvisionerConfig{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			// We want to trigger the reconciliation of SpaceProvisionerConfigs that reference some ToolChainCluster whenever
+			// the ToolChainClusters are created or deleted. We don't have to care about updates, the mere existence of the
+			// ToolchainCluster is enough for us.
+			&source.Kind{Type: &toolchainv1alpha1.ToolchainCluster{}},
+			handler.EnqueueRequestsFromMapFunc(MapToolChainClusterToSpaceProvisionerConfigs(ctx, r.Client)),
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(event.CreateEvent) bool {
+					return true
+				},
+				DeleteFunc: func(event.DeleteEvent) bool {
+					return true
+				},
+			},
+			)).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaceprovisionerconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaceprovisionerconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=toolchainclusters,verbs=get;list;watch
+
+// Reconcile ensures that SpaceProvisionerConfig is valid and points to an existing ToolchainCluster.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	spaceProvisionerConfig := &toolchainv1alpha1.SpaceProvisionerConfig{}
+	if err := r.Client.Get(ctx, request.NamespacedName, spaceProvisionerConfig); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("SpaceProvisionerConfig not found anymore")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get SpaceProvisionerConfig: %w", err)
+	}
+
+	if util.IsBeingDeleted(spaceProvisionerConfig) {
+		logger.Info("SpaceProvisionerConfig is being deleted - skipping reconciliation")
+		return ctrl.Result{}, nil
+	}
+
+	// check that there exists a ToolchainCluster CR in the same namespace
+	toolchainCluster := &toolchainv1alpha1.ToolchainCluster{}
+	toolchainClusterKey := runtimeclient.ObjectKey{Name: spaceProvisionerConfig.Spec.ToolchainCluster, Namespace: spaceProvisionerConfig.Namespace}
+	toolchainPresent := corev1.ConditionTrue
+	toolchainPresenceReason := toolchainv1alpha1.SpaceProvisionerConfigValidReason
+	if err := r.Client.Get(ctx, toolchainClusterKey, toolchainCluster); err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to get ToolchainCluster: %w", err)
+		}
```

Comment:
Let's set the status when it fails for any type of error. We don't know what it failed and it's always good to have the visibility in the status of the CR. 
in this case, we can also include the content of the error as the message in the condition - something like this:
```go
	message := ""
	if err := r.Client.Get(ctx, toolchainClusterKey, toolchainCluster); err != nil {
		toolchainPresenceReason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason
		toolchainPresent = corev1.ConditionFalse
		message = err.Error()
...
		toolchainv1alpha1.Condition{
			Type:    toolchainv1alpha1.ConditionReady,
			Status:  toolchainPresent,
			Reason:  toolchainPresenceReason,
			Message: message,
		})
```

Reply:
done, but please take a look, because I did it slightly differently to what you proposed here.